kategory/src/main/kotlin/kategory/data/SequenceKW.kt

```diff
@@ -0,0 +1,84 @@
+package kategory
+
+fun <A> SequenceKWKind<A>.toList(): List<A> = this.ev().sequence.toList()
+
+@higherkind
+@deriving(Monad::class, Traverse::class, MonoidK::class)
+data class SequenceKW<out A> constructor(val sequence: Sequence<A>) : SequenceKWKind<A>, Sequence<A> by sequence {
+
+    fun <B> flatMap(f: (A) -> SequenceKWKind<B>): SequenceKW<B> = this.ev().sequence.flatMap { f(it).ev().sequence }.k()
+
+    fun <B> map(f: (A) -> B): SequenceKW<B> = this.ev().sequence.map(f).k()
+
+    fun <B> foldL(b: B, f: (B, A) -> B): B = this.ev().fold(b, f)
+
+    fun <B> foldR(lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> {
+        fun loop(fa_p: SequenceKW<A>): Eval<B> = when {
+            fa_p.sequence.none() -> lb
+            else -> f(fa_p.ev().sequence.first(), Eval.defer { loop(fa_p.sequence.drop(1).k()) })
```

Comment:
one question also to @pakoito, 

1. Why not use `fa_p.ev().first()` or `fa_p.drop(1).k()` instead ?
2. Why do you use `ev()` method only for `fa_p.ev().sequence.first()` and not for `drop`? 
3. Is it necessary to use evidence method (`ev()`) here?

Reply:
i think `f(fa_p.first(), Eval.defer { loop(fa_p.drop(1).k()) })` works as well. i don't have a strong opinion on this. up to you guys

Reply:
Yes, it's works because you are delegating all public `val sequence: Sequence<A>` methods to `SequenceKW`

Reply:
Nits mostly, if @pt2121 wants I'll just go ahead and merge.

Reply:
i can do that. gimme a couple minutes